src/drivers/dshot/DShot.cpp

```diff
@@ -31,257 +31,47 @@
-	int sendCommandThreadSafe(dshot_command_t command, int num_repetitions, int motor_index);
-
-	void retrieveAndPrintESCInfoThreadSafe(int motor_index);
-
-	bool telemetryEnabled() const { return _telemetry != nullptr; }
-
-private:
-
-	void Run() override;
-
-	static constexpr uint16_t DISARMED_VALUE = 0;
-
-	enum class DShotConfig {
-		Disabled = 0,
-		DShot150 = 150,
-		DShot300 = 300,
-		DShot600 = 600,
-		DShot1200 = 1200,
-	};
-
-	struct Command {
-		dshot_command_t command;
-		int num_repetitions{0};
-		uint8_t motor_mask{0xff};
-
-		bool valid() const { return num_repetitions > 0; }
-		void clear() { num_repetitions = 0; }
-	};
-
-	struct Telemetry {
-		DShotTelemetry handler;
-		uORB::PublicationData<esc_status_s> esc_status_pub{ORB_ID(esc_status)};
-		int last_motor_index{-1};
-	};
+#include "DShot.h"
 
-	void updateTelemetryNumMotors();
-	void initTelemetry(const char *device);
-	void handleNewTelemetryData(int motor_index, const DShotTelemetry::EscData &data);
+char DShot::_telemetry_device[] {};
+px4::atomic_bool DShot::_request_telemetry_init{false};
 
-	int requestESCInfo();
-
-	MixingOutput _mixing_output{DIRECT_PWM_OUTPUT_CHANNELS, *this, MixingOutput::SchedulingPolicy::Auto, false, false};
-
-	Telemetry *_telemetry{nullptr};
-	static char _telemetry_device[20];
-	static px4::atomic_bool _request_telemetry_init;
-
-	px4::atomic<DShotTelemetry::OutputBuffer *> _request_esc_info{nullptr};
-	bool _waiting_for_esc_info{false};
-
-	Mode		_mode{MODE_NONE};
-
-	uORB::Subscription _param_sub{ORB_ID(parameter_update)};
-
-	Command _current_command;
-	px4::atomic<Command *> _new_command{nullptr};
-
-	unsigned	_num_outputs{0};
-	int		_class_instance{-1};
-
-	bool		_outputs_on{false};
-	uint32_t	_output_mask{0};
-	bool		_outputs_initialized{false};
-
-	perf_counter_t	_cycle_perf;
-
-	void		capture_callback(uint32_t chan_index,
-					 hrt_abstime edge_time, uint32_t edge_state, uint32_t overflow);
-	int			pwm_ioctl(file *filp, int cmd, unsigned long arg);
-	void		update_dshot_out_state(bool on);
-
-	void		update_params();
-
-	int		capture_ioctl(file *filp, int cmd, unsigned long arg);
-
-	DShotOutput(const DShotOutput &) = delete;
-	DShotOutput operator=(const DShotOutput &) = delete;
-
-	DEFINE_PARAMETERS(
-		(ParamInt<px4::params::DSHOT_CONFIG>) _param_dshot_config,
-		(ParamFloat<px4::params::DSHOT_MIN>) _param_dshot_min,
-		(ParamInt<px4::params::MOT_POLE_COUNT>) _param_mot_pole_count
-	)
-};
-
-char DShotOutput::_telemetry_device[] {};
-px4::atomic_bool DShotOutput::_request_telemetry_init{false};
-
-DShotOutput::DShotOutput() :
+DShot::DShot() :
 	CDev("/dev/dshot"),
-	OutputModuleInterface(MODULE_NAME, px4::wq_configurations::hp_default),
-	_cycle_perf(perf_alloc(PC_ELAPSED, MODULE_NAME": cycle"))
+	OutputModuleInterface(MODULE_NAME, px4::wq_configurations::hp_default)
 {
-	_mixing_output.setAllDisarmedValues(DISARMED_VALUE);
-	_mixing_output.setAllMinValues(DISARMED_VALUE + 1);
+	_mixing_output.setAllDisarmedValues(DSHOT_DISARM_CMD);
```

Comment:
Can you change this to `DSHOT_DISARM_VALUE` / `DSHOT_DISARM_THROTTLE` to avoid potential confusion?

Reply:
Certainly. Done. Rebased against current master, commits squashed and pushed up.